### PR TITLE
Follow up to fix color picker on wallet

### LINKED ISSUE #17443: Make part of next color picker visible in color pickers scroll bar
Followup of https://github.com/status-im/status-mobile/pull/17405 cc @OmarBasem 

Discussed [here](https://discord.com/channels/624634427930312714/928625369542713396/1156865341092274256)

Currently part of next color picker is not visible wich makes it not obvious for user that color pickers are scrollable. 

We should make part of next color picker visible as it is shown on the designs 

**Expected result:** 

https://www.figma.com/file/o4qG1bnFyuyFOvHQVGgeFY/Onboarding-for-Mobile?type=design&node-id=2241-250931&mode=design

Onboarding for Mobile – Figma 2023-09-28 11-35-17

**Actual result:** 

photo_2023-09-28 11 07 33

### Additional Information

- Status version: nightly 
- Operating System: Android, iOS

### PR DESCRIPTION
**Pull request by ibrkhalil**
fixes #17443 followup

### Summary

Fixes a regression caused by 17443 on create wallet screen.
image

status: ready

### CODE REVIEW

**Comment by J-Son89** [comment]
thanks @ibrkhalil! I should have also checked this on your last pr too 👍

File: src/utils/responsiveness.cljs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+(ns utils.responsiveness)
+
+(def ^:const IPHONE_11_PRO_VIEWPORT_WIDTH 375)
+
+(defn iphone-11-Pro-20-pixel-from-width
```
Comment (J-Son89) [inline src/utils/responsiveness.cljs:5]: 👍

Comment (J-Son89) [reply src/utils/responsiveness.cljs:5]: Personally I would put the comments on top of the function to explain it's use and not inside. like a docstring type comment

**Review by J-Son89 — APPROVED** [review]

**Comment by pavloburykh** [comment]
> thanks @ibrkhalil! I should have also checked this on your last pr too 👍

That's my bad. I have checked on nightly build by mistake and missed that wallet color pickers were totally broken the PR. Based on wrong check I assumed that wallet colour pickers just do not have dynamic margin and thought that it is fine to fix in a followup...

**Comment by pavloburykh** [comment]
@ibrkhalil thank you! Wallet color pickers are not broken any more. But still the last hidden picker differs from Figma. Will it be adjusted by wallet team or wallet color pickers should look correct in this PR? 

### ISSUE 1 Less than half of the colorpicker is hidden by default

**Expected result:** half of the colorpicker should be hidden 

[Figma](https://www.figma.com/file/HKncH4wnDwZDAhc4AryK8Y/Wallet-for-Mobile?type=design&node-id=1-132380&mode=design&t=D6Fa3XptBxJuyzvR-0) 

Wallet for Mobile – Figma 2023-10-30 12-17-59

**Actual result:** less than half of the colorpicker is hidden

photo_2023-10-30 12 14 09

**Comment by pavloburykh** [comment]
@ibrkhalil also, there is another regression: 

### ISSUE 2 Ring around selected color is lost

**Expected result:** 

[Figma](https://www.figma.com/file/HKncH4wnDwZDAhc4AryK8Y/Wallet-for-Mobile?type=design&node-id=1-132380&mode=design&t=D6Fa3XptBxJuyzvR-0)

Wallet for Mobile – Figma 2023-10-30 12-10-22

**Actual result:**

photo_2023-10-30 12 22 24

**Comment by ibrkhalil** [comment]
> @ibrkhalil also, there is another regression:
> 
> ### ISSUE 2 Ring around selected color is lost
> **Expected result:**
> 
> [Figma](https://www.figma.com/file/HKncH4wnDwZDAhc4AryK8Y/Wallet-for-Mobile?type=design&node-id=1-132380&mode=design&t=D6Fa3XptBxJuyzvR-0)
> 
> Wallet for Mobile – Figma 2023-10-30 12-10-22
> **Actual result:**
> 
> photo_2023-10-30 12 22 24

Fixed :)

File: src/status_im2/contexts/wallet/create_account/view.cljs [hunk 2]
```diff
@@ -93,8 +95,9 @@
         [quo/color-picker
          {:default-selected @account-color
           :on-change        #(reset! account-color %)
-          :container-style  {:padding-horizontal 12
-                             :padding-vertical   12}}]]
+          :window-width     window-width
```
Comment (J-Son89) [inline src/status_im2/contexts/wallet/create_account/view.cljs]: imo this prop should just be `width`. 
also can we consider just defaulting this inside the component so we can overwrite it but we don't need to pass this each use as it's common for all screens using this? 🤔

**Comment by pavloburykh** [comment]
> Fixed :)

Thank you @ibrkhalil 

Could you please also clarify regarding [ISSUE 1](https://github.com/status-im/status-mobile/pull/17748#issuecomment-1784888223)?

**Comment by ibrkhalil** [comment]
> > Fixed :)
> 
> Thank you @ibrkhalil
> 
> Could you please also clarify regarding [ISSUE 1](https://github.com/status-im/status-mobile/pull/17748#issuecomment-1784888223)?

Sorry Pavlo for not being clear
I meant both are done.

File: src/utils/responsiveness.cljs [hunk 3]
```diff
@@ -0,0 +1,10 @@
+(ns utils.responsiveness)
+
+(def ^:const IPHONE_11_PRO_VIEWPORT_WIDTH 375)
```
Comment (OmarBasem) [inline src/utils/responsiveness.cljs:3]: I don't think a specific device width needs to be in the calculation even if that's what we are building to. 

We just need the current device's width, and each circle's width, and then we can calculate if a part of the last circle will be visible or not

Comment (ibrkhalil) [reply src/utils/responsiveness.cljs:3]: I want the look to be exactly the same as Figma on 11 Pro and as close as possible for other devices.

Comment (OmarBasem) [reply src/utils/responsiveness.cljs:3]: We still don't need iphone 11 width. On figma we have 6 and a half circles visible, and 7 spaces for margin. If the device width is `X` and cirlce's width is `Y`, then we have the total margin to be `X - (6.5 Y)`, let's call that value `Z`. Therefore, one margin is `Z / 7`.

Comment (ibrkhalil) [reply src/utils/responsiveness.cljs:3]: That'd work too, But I still think having a reference to Figma's dimension is a better approach.

**Comment by pavloburykh** [comment]
@ibrkhalil thank you for the fixes. No issues from my side. Ready for merge.

File: src/status_im2/contexts/wallet/create_account/view.cljs (hunk 2 above)
Comment (ibrkhalil) [reply src/status_im2/contexts/wallet/create_account/view.cljs]: Done

**Pull request merged**